### ISSUE TITLE
Pyenv loads grammar files from system Python, not from the version it installed

### ISSUE THREAD

**Issue by PetterS** [pyenv/pyenv#1734]
This is my test script:
```
from lib2to3 import pygram
from lib2to3 import pytree
from lib2to3.pgen2 import driver

driver = driver.Driver(pygram.python_grammar_no_print_statement, convert=pytree.convert)
driver.parse_string("if x := 1: pass\n", debug=False)
print("OK")
```

On one computer, where the system Python is 3.8.5, it prints "OK" as expected.

On another system where the system Python is 3.6.9, it crashes when pyenv executes the script with python 3.8.2:
```
Traceback (most recent call last):
  File "<path redacted>/petter.py", line 6, in <module>
    driver.parse_string("if x := 1: pass\n", debug=False)
  File "/home/<user redacted>/.pyenv/versions/3.8.2/lib/python3.8/lib2to3/pgen2/driver.py", line 103, in parse_string
    return self.parse_tokens(tokens, debug)
  File "/home/<user redacted>/.pyenv/versions/3.8.2/lib/python3.8/lib2to3/pgen2/driver.py", line 71, in parse_tokens
    if p.addtoken(type, value, (prefix, start)):
  File "/home/<user redacted>/.pyenv/versions/3.8.2/lib/python3.8/lib2to3/pgen2/parse.py", line 162, in addtoken
    raise ParseError("bad input", type, value, context)
lib2to3.pgen2.parse.ParseError: bad input: type=22, value='=', context=('', (1, 6))
```
Note that pyenv runs the correct Python files, but the loaded grammar is for Python 3.6.

TL;DR: To reproduce the bug:
- Have a system with Python 3.6.
- Install Python 3.8 with pyenv.
- Run the script with Python 3.8.

**Comment by PetterS** [comment]
Here is the grammar file in Cpython: https://github.com/python/cpython/blob/master/Lib/lib2to3/pygram.py#L15

Looks good, it is a relative path. But what if the file is missing? This code is repsonsible for loading: https://github.com/python/cpython/blob/c79667ff7921444911e8a5dfa5fba89294915590/Lib/lib2to3/pgen2/driver.py#L143

It does some other heuristics for loading the grammar if the file is missing.

**Comment by PetterS** [comment]
I should add that this bug breaks `yapf` so that it can not format files with `:=` in them, so it is a fairly serious bug.

**Comment by PetterS** [comment]
This was a bug in Python 3.8.2: https://github.com/google/yapf/issues/787
